4.applications/grit-informed-aggregation/1.generate-profiles/scripts/nbconverted/2.calculate-replicate-reproducibility.py

```diff
@@ -0,0 +1,258 @@
+
+# In[2]:
+
+
+import os
+import glob
+import gzip
+from pathlib import Path
+
+import random
+import numpy as np
+import pandas as pd
+from scipy.special import softmax
+
+import matplotlib.pyplot as plt
+import seaborn as sns
+import plotnine as gg
+
+from pycytominer.cyto_utils import infer_cp_features
+from cytominer_eval import evaluate
+from scripts.utils import calculate_weighted_agg
+import scripts.eval_utils as utils
+
+
+# In[3]:
+
+
+input_folder = "data/aggregated-profiles/"
+results_folder = "results/"
+
+
+# In[4]:
+
+
+get_ipython().run_cell_magic(
+    "time",
+    "",
+    "\ndf_list = [pd.read_csv(file, sep='\\t') for file in glob.glob(input_folder+'*.tsv')] # for the FULL set of normalized profiles\nconcat_df = pd.concat(df_list, axis='rows', sort=False)\nprint(concat_df.shape)\n\n# save the results\n# replicate correlation at the guide-level\nf_guide_cor_list = []\nf_guide_df_list = []\n# replicate correlation at the gene-level\nf_gene_cor_list = []\nf_gene_df_list = []\n# method list\nmethod_list = []\n# cell_line list\ncell_list = []\n\nfor method in concat_df.Metadata_agg_method.unique():\n    for cell_line in concat_df.Metadata_cell_line.unique():\n        print(f\"method is {method}, cell line is {cell_line}\")\n        sub_df = concat_df.query(\"Metadata_agg_method == @method & Metadata_cell_line == @cell_line\")\n\n        # guide-level replicate correlation\n        f_guide_cor_val, f_guide_cor_df = evaluate(\n            profiles = sub_df,\n            features = infer_cp_features(sub_df),\n            meta_features = infer_cp_features(sub_df, metadata=True),\n            replicate_groups=['Metadata_pert_name', 'Metadata_gene_name', 'Metadata_cell_line', 'Metadata_agg_method'], \n            operation = 'replicate_reproducibility',\n            similarity_metric = 'pearson',\n        #         replicate_reproducibility_quantile = 0.99,\n            replicate_reproducibility_return_median_cor = True\n                                  )\n        f_guide_cor_list.append(f_guide_cor_val)\n        f_guide_df_list.append(f_guide_cor_df)    \n    \n        # gene-level replicate correlation\n        f_gene_cor_val, f_gene_cor_df = evaluate(\n            profiles = sub_df,\n            features = infer_cp_features(sub_df),\n            meta_features = infer_cp_features(sub_df, metadata=True),\n            replicate_groups=['Metadata_gene_name', 'Metadata_cell_line', 'Metadata_agg_method'],\n            operation = 'replicate_reproducibility',\n            similarity_metric = 'pearson',\n        #         replicate_reproducibility_quantile= 0.99,\n            replicate_reproducibility_return_median_cor = True\n                                  )\n        f_gene_cor_list.append(f_gene_cor_val)\n        f_gene_df_list.append(f_gene_cor_df)\n        \n        method_list.append(method)\n        cell_list.append(cell_line)\n\nguide_rep_cors = pd.concat(f_guide_df_list)\ngene_rep_cors = pd.concat(f_gene_df_list)",
```

Comment:
should these files be normalized feature selected files instead of just normalized?

We talked about subsetting the features to those in a specific file in this repo:

> use these features: https://github.com/broadinstitute/grit-benchmark/tree/main/1.calculate-metrics/cell-health/data

Do you think we should use these here as well?

Reply:
consistency is the name of the game

Reply:
done in 335d81483e54227b0b9d61373582b124ee1dba2e!